pints/_mcmc/_slice_hyperrectangles.py

```diff
@@ -0,0 +1,283 @@
+# -*- coding: utf-8 -*-
+#
+# Hyperrectangles-based Slice Sampling
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+
+
+class SliceHyperrectanglesMCMC(pints.SingleChainMCMC):
+    """
+    Implements Hyperrectangles-based Slice Sampling, as described in [1].
+
+    This is a multivariate method, which generates n-dimensional samples of
+    the form ``x = (x_1, ..., x_n)`` by sampling uniformly from the area of an
+    axis-aligned hyperrectangle:
+    ``H = {x: L_i < x_i < R_i for all i = 1, ..., n}``.
+    Here, ``L_i`` and ``R_i`` define  the extent of the hyperrectangle along
+    the ``i`` th axis.
+
+    Sampling follows:
+
+    1. Calculate the pdf (``f(x0)``) of the current sample (``x0``).
+    2. Draw a real value (``y``) uniformly from (0, f(x0)), defining a
+       horizontal “slice”: S = {x: y < f (x)}. Note that ``x0`` is
+       always within S.
+    3. Find a hyperrectangle (``H = (L_1, R_1) ×···× (L_n, R_n)``) around
+       ``x_0``, which preferably contains at least a big part of the slice.
+    4. Draw a new point (``x1``) from the part of the slice within this
+       hyperrectangle.
+
+    The implementation uses estimates (``w_i``) of the relative scales of the
+    variables to randomly position a hyperrectangle with such dimensions
+    uniformly over positions containing ``x_0`` that lead to ``H``. The
+    algorithm consists of the following steps, as described in [1] Fig. 8.
+    pp.723:
+
+    1. ``y \sim uniform(0, f(x_0))``
+    2. for ``i = 1`` to ``n``:
+        a. ``U_i \sim uniform(0,1)``
+        b. ``L_i = x_{0_i} - w_i U_i``
+        c. ``L_i + w_i``
```

Comment:
I don't understand step 2c. Why are we just adding these two variables? Surely a variable must be being set here?

Reply:
From the code, think it is R_i being set here - can you change?